govcd/lbserverpool_test.go

```diff
@@ -0,0 +1,123 @@
+// +build lb lbServerPool functional ALL
+
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+// Test_LBServerPool tests CRUD methods for load balancer server pool.
+// The following things are tested if prerequisite Edge Gateway exists:
+// Creation of load balancer server pool
+// Read load balancer server pool by both ID and Name (server pool name must be unique in single edge gateway)
+// Update - change a single field and compare that configuration and result objects are deeply equal
+// Update - try and fail to update without mandatory field
+// Delete
+func (vcd *TestVCD) Test_LBServerPool(check *C) {
```

Comment:
I am not fond of such unit test where all functionality tested. It more like workflow test and not function unit test. @dataclouder

Reply:
I could split them, however this would make total run time longer due to for situations such as Test_ReadLBServerPool, Test_UpdateLBServerPool, Test_DeleteLBServerPool I would still need to create for  each scenario.